Tasks/Maven/maventask.ts

```diff
@@ -21,16 +21,16 @@ if (mavenVersionSelection == 'Path') {
 tl.debug('Maven binary: ' + mvntool);
 
 var mavenPOMFile = tl.getPathInput('mavenPOMFile', true, true);
-var mavenOptions = tl.getDelimitedInput('options', ' ', false);
-var mavenGoals = tl.getDelimitedInput('goals', ' ', true);
+var mavenOptions = tl.getInput('options', false); // options can have spaces and quotes so we need to treat this as one string and not try to parse it
+var mavenGoals = tl.getDelimitedInput('goals', ' ', true); // This assumes that goals cannot contain spaces
 
 var mvnv = tl.createToolRunner(mvntool);
 mvnv.arg('-version');
 
 var mvnb = tl.createToolRunner(mvntool);
 mvnb.arg('-f');
 mvnb.pathArg(mavenPOMFile);
-mvnb.arg(mavenOptions);
+mvnb.argString(mavenOptions);
```

Comment:
yep.  argString parses between double quotes and supports escaping a double quote as well.